securitycenter/securitymarksv2/assets_test.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package securitymarksv2
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	securitycenter "cloud.google.com/go/securitycenter/apiv1"
+	"cloud.google.com/go/securitycenter/apiv1/securitycenterpb"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"google.golang.org/api/iterator"
+	"google.golang.org/genproto/protobuf/field_mask"
+)
+
+var marksAssetName = ""
+
+func getRandomAsset(client *securitycenter.Client, orgID string) (*securitycenterpb.Asset, error) {
+	ctx := context.Background()
+	req := &securitycenterpb.ListAssetsRequest{
+		Parent: fmt.Sprintf("organizations/%s", orgID),
+	}
+
+	var randomAsset *securitycenterpb.Asset
+	assetsCount := 0
+	it := client.ListAssets(ctx, req)
+	for {
+		result, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("it.Next(): %w", err)
+		}
+		assetsCount++
+		if rand.Float64() < 1.0/float64(assetsCount) {
+			randomAsset = result.Asset
+		}
+	}
+	return randomAsset, nil
+}
+
+func attemptLease(client *securitycenter.Client, asset *securitycenterpb.Asset, orgID string) error {
+	const leaseExpirationKey = "LEASEKEY"
+	now := time.Now().UnixNano()
+
+	lease := asset.SecurityMarks.Marks[leaseExpirationKey]
+	okToLease := true
+	if lease != "" {
+		i, err := strconv.ParseInt(lease, 10, 64)
+		if err != nil {
+			fmt.Printf("strconv.ParseInt(%v, 10, 64): %v", lease, err)
```

Comment:
+1. Also (minor nit): for printing the error perhaps stringify the argument ("%q" instead of "%v" would look better and be consistent with other recommendations below).

Reply:
Thank you for pointing it out. Addressed.